front/lib/api/assistant/get_favorite_states.ts

```diff
@@ -0,0 +1,33 @@
+import { Op } from "sequelize";
```

Comment:
what about adding it to `user_relation.ts` where there is already `setAgentUserFavorite`?
also, out of consistency, would name it similarly `getAgentUserFavorite` (and plural)

Reply:
> getAgentUserFavorites

Will change (i just find getAgentUserFavorites to be confusing with Agent singular and the double plural isn't great either).

Reply:
> I don't understand, where would the circular dependency be?

Sorry, I say "I use" but I meant: "we use getAgentLightConfiguration() in setAgentUserFavorite()".
So when I added my function to the same file then `front/lib/api/assistant/configuration.ts` imported `front/lib/api/assistant/user_relation.ts` (to get the function to retrieve favorites states) who imported `front/lib/api/assistant/configuration.ts`.

Reply:
Ok I understand :)

IMO it is worth it to have symmetric functions in the same file
`setAgentUFavorite` imports `getAgentConfiguration` solely to check for existence and agent status (active)

=> if we beleive those checks are legitimate, then they should be done for the getter too, and thus rework configuration.ts (or optionally make the check via direct DB call rather than importing get) => and both can be in the same file

=> if we beleive those checks are actually not needed, then we can drop them from the setter so the import goes away and both can be in the same file too :)

if you agree, all above solutions are fine IMO, so trusting you to pick the best

Reply:
:+1: honestly i'd rather kill the checks than have 2 separate files (but ofc the best would be if the refactor to have both the checks are fast)
ready to help if you'd like

Reply:
I'm doing something else right now, feel free to take the branch and update :)

Reply:
@philipperolet 

I took some more time to look at it today and I don't feel comfortable removing the checks (they do test a lot of things to make sure this agent can be used by the user).

The other option I was comptemplating was to put the 2 favorites functions in the `/lib/assistant/configurations.ts` file but it lead to another cycle dep (`global_agent.ts` -> `configurations.ts` -> `global_agent.ts`).

This is clearly a rabbit hole :) => merging as it.